workers/dockerfiles/Dockerfile.salmon

```diff
@@ -23,41 +30,29 @@ WORKDIR /home/user
 
 # Install Salmon
 ENV SALMON_VERSION 0.9.1
-
 RUN wget https://github.com/COMBINE-lab/salmon/releases/download/v${SALMON_VERSION}/Salmon-${SALMON_VERSION}_linux_x86_64.tar.gz
-
 RUN tar -xzf Salmon-${SALMON_VERSION}_linux_x86_64.tar.gz
-
-# Salmon can extract to a different directory than the name of the tar file.
```

Comment:
The main reason for this change: when the pre-compiled salmon library files were copied into /usr/local/lib/, they have higher priorities than the library files in `/usr/lib/`, and messed up both JRE and Rscript. Alternatively, we can compile the source code of `salmon`, but they requires boost library, which makes the docker at least 150MB image larger. So I tweaked the PATH (see below) instead.

Reply:
Wait I'm confused, I don't actually see you tweaking the PATH at all. It looks like you're making `/usr/local/bin` into a softlink which points to the `bin` that came with salmon? It seems like if you wanted to add salmon's bin to the PATH you could just append it to the `$PATH` environment variable right? It also seems like another potential solution could be change `$PATH` to prioritize `/usr/lib` over `/usr/local/lib`?

No matter how we end up doing this, I do think the explanation should be a comment in the code instead of on the PR so that we can see it if we ever go to modify this later.

Reply:
Originally I played with the $PATH, then I realized that no matter $PATH is defined in ~/.bashrc or ~/.profile, it will be effective only in BASH, so adding a soft link in `/usr/local/bin/` seems better. (Generally it is not a good idea to tweak the dynamic library path, because it may affect other programs.)

Reply:
I will add a comment line in this file.